From 93e9a3a2a20ee6d0fe74c67c992f22adf85bf26e Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Fri, 17 May 2019 18:08:21 +0200
Subject: [PATCH 241/386] PCI: Make some quirqs dependent on PCI_IOV

On some systems where the PCI_IOV is disabled some quirqs are not able to
compile - make them PCI_IOV dependent.

Change-Id: I8accc4739202f2fa0a5c88b423bb21c7ddff9986
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/9174
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stanislaw Kardach <Stanislaw.Kardach@cavium.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/pci/quirks.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/pci/quirks.c b/drivers/pci/quirks.c
index 01ff9bf29cf6..5b15853b26e2 100644
--- a/drivers/pci/quirks.c
+++ b/drivers/pci/quirks.c
@@ -1011,6 +1011,7 @@ static void quirk_cavium_sriov_rnm_link(struct pci_dev *dev)
 DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_CAVIUM, 0xa018, quirk_cavium_sriov_rnm_link);
 #endif
 
+#ifdef CONFIG_PCI_IOV
 /*
  * Cavium OcteonTx needs Multibyte atomic I/O (LDST) for some devices.
  * LDST needs LMTLINE and LMTCANCEL to be mapped along with devices,
@@ -1090,6 +1091,7 @@ static void quirk_octeontx_ssombox(struct pci_dev *dev)
 	dev_info(&dev->dev, "quirk(SSO MBOX): added at BAR 4\n");
 }
 DECLARE_PCI_FIXUP_HEADER(PCI_VENDOR_ID_CAVIUM, 0XA04C, quirk_octeontx_ssombox);
+#endif
 
 /*
  * Some settings of MMRBC can lead to data corruption so block changes.
-- 
2.17.1

